src/components/crown/crownConfig/crownConfig.vue

```diff
@@ -134,6 +138,12 @@ export default {
     },
   },
   methods: {
+    closeBoundaryEventDropDown() {
+      this.$refs.boundaryEventDropdown.dropdownOpen = false;
+    },
+    closeTaskDropDown() {
+      this.$refs.taskDropdown.dropdownOpen = false;
```

Comment:
This is my opinion, but I want to discourage the use of `$refs` as much as possible. They are an "escape hatch" that should really only be used for accessing native DOM attributes.

I consider it an escape hatch because a component has a well-defined API: props and events. When using a ref, this API is sidestepped and it'll make refactoring and reasoning about the component more difficult.

`dropdownOpen` is part of the component's internal state, and should really only be modified through props. If we want to refactor either of the dropdown components, we now have to consider all the places where the component might be used and determine if a ref is relying on any of the components internal details.

My suggestion is to use props to share the dropdown open status between the two components. Let me know what you think.

Reply:
Makes sense. If the data prop was renamed or removed, you can't directly see the association.